index.js

```diff
@@ -1,79 +1,77 @@
+/* global -Promise */
+var Promise = require('es6-Promise').Promise;
 
-function FindApp (opts, callback) {
+module.exports = findApp;
 
-  opts = opts ? __.clone(opts) : {};
+function findApp(options) {
+  
+  options = options || {};
 
-  var manifestJSON = getManifest(opts.manifestURL);
-  var webapps = getWebapps(opts.client);
-  var apps = webapps.then(getInstalledApps);
+  var client = options.client;
+  var manifest = options.manifest;
+
+  return new Promise(function(resolve, reject) {
+
+    if (client === undefined || manifest === undefined) {
+      return reject(new Error('We need a client and a manifest to find an app'));
+    }
+
+    getWebAppsActor(client)
+      .then(listInstalledApps)
+      .then(filterAppsByName(manifest.name))
+      .then(function(result) {
+        resolve(result);
+      });
 
-  var appManifest = Q.all([manifestJSON, apps]).spread(findAppManifest);
-  var appActor = appManifest.then(function(app) {
-    return webapps.then(getApp(app.manifestURL));
   });
 
-  // Fulfilling the appManifest means that the app is installed
-  // Fulfilling the appActor means that the app is running
-  return Q.allSettled([appManifest, appActor])
-    .spread(function(manifest, actor) {
+}
 
-      // No manifest means no app installed
-      if (manifest.state == 'rejected') {
-        throw manifest.reason;
-      }
 
-      var result = {};
+function getWebAppsActor(client) {
+  return new Promise(function(resolve, reject) {
 
-      if (actor.state == 'fulfilled') {
-        result = actor.value;
+    client.getWebapps(function(err, webapps) {
+      if (err) {
+        return reject(err);
       }
 
-      result.manifest = manifest.value;
-      return result;
-    })
-    .nodeify(callback);
-}
+      resolve(webapps);
 
-function getLocalPath(styleActor) {
-  var matches = REMOTE_PATH.exec(styleActor.sheet.href);
-  return matches[1];
-}
+    });
 
-function getApp (manifestURL) {
-  return function(webapps) {
-    return Q.ninvoke(webapps, 'getApp', manifestURL);
-  };
+  });
 }
 
-function getWebapps(client) {
-  return Q.ninvoke(client, 'getWebapps');
-}
 
-function getInstalledApps(webapps) {
-  return Q.ninvoke(webapps, 'getInstalledApps');
-}
+function listInstalledApps(webAppsActor) {
+  return new Promise(function(resolve, reject) {
+
+    webAppsActor.getInstalledApps(function(err, apps) {
+
+      if (err) {
+        return reject(err);
+      }
+
+      resolve(apps);
 
-function getManifest(manifestURL) {
-  return Q.nfcall(fs.readFile, manifestURL, 'utf8')
-    .then(function(file) {
-      return JSON.parse(file);
     });
+
+  });
 }
 
-function findAppManifest(manifest, apps) {
-  for (var i=0; i < apps.length; i++) {
-    var app = apps[i];
-    if (app.name == manifest.name) {
-      return app;
-    }
-  }
-  throw new Error("App not found");
-}
\ No newline at end of file
+
+function filterAppsByName(name) {
+
+  return function(apps) {
```

Comment:
I've rewritten that part! Let me know how it looks now :)
Perhaps it was a bit too clever, but we actually didn't need the Promise there on second thoughts. I got carried away :)